rails_programming/hotwire/turbo.md

```diff
@@ -0,0 +1,494 @@
+~~~
+
+which will generate:
+
+~~~html
+<a href="..." data-turbo-method="delete">Delete Article</a>
+~~~
+
+This creates a link that will use the `DELETE` method. However, it is suggested that you use a button or form for anything that isn't a `GET` request
+
+#### Disable Turbo Drive
+
+There will be times where you do not want Turbo Drive to control navigation and want a full page reset.
+
+You can disable Turbo Drive by adding `data-turbo="false` directly on your links or on the parent containing them.
+
+For instance:
+
+~~~erb
+<div data-turbo="false">
+  <%= link_to "foo", "bar" %>
+  <%= link_to "baz", "qux", data: { turbo: "true" } %>
+</div>
+~~~
+
+In the above example, we created a parent div with `data-turbo="false"`, which disables Turbo Drive on all elements inside of it. However, we also added `data: { turbo: "true" }` to the second `link_to`. This will turn Turbo Drive back on for that particular element.
+
+This showcases that you can disable and re-enable Turbo Drive in regions of your website and select specific elements to enable or disable it on. Remember that when using ERB tags (like `link_to`) we use the syntax style of `data: { foo: "bar" }` rather than the HTML style of `data-foo="bar"`
+
+### Turbo Frames
+
+#### Creating a Frame
+
+Imagine a piece of paper and cutting out a small square hole in it. You could change what you see through the hole by swapping out another piece of paper behind it, but the rest of the paper will always look the same. That's the idea of Turbo Frames! Turbo Frames allow us to predefine a portion of our page to be replaced during a request. Any links or forms inside of our frame will make a special request that results in only changing the frame. A page can also have multiple Turbo Frames!
+
+A frame is designated by wrapping a region inside of a `<turbo-frame>` element. Rails has a special tag for this, `<%= turbo_frame_tag %>`.
+
+A basic Turbo Frame, using Rails helpers, may look like so:
+
+~~~erb
+<%= turbo_frame_tag "article" do %>
+  Some content
+<% end %>
+~~~
+
+which will generate:
+
+~~~html
+<turbo-frame id="article">
+  Some content
+</turbo-frame>
+~~~
+
+Super easy! Note that the frames have an ID. The ID is how Turbo is able to identify a frame to find out which one is which.
+With the Turbo Frame helper, you can substitute the ID for a variable. For instance:
+
+~~~erb
+<% @articles.each do |article| %>
+  <%= turbo_frame_tag article %>
+    <%= article.title %>
+  <% end %>
+<% end %>
+~~~
+
+The above example will generate a turbo frame for every article. Each frame will have a unique id like `article_1` or `article_2` and all we had to include was our article variable! 
+
+#### Connecting to Other Frames
+
+Now that we have our first frame, we need to connect it to another frame to swap the content out! This is very simple, all we have to do is create a link inside of a Turbo Frame where the requested view *also* has a Turbo Frame with the **same ID**.
```

Comment:
I find this formulation a bit confusing: 

> Now that we have our first frame, we need to connect it to another frame to swap the content out!

Not sure if `connecting` is misleading. How about something like:
```md
#### Updating the content of a turbo frame

Now that we have our first frame, we can replace its content with a link that request new frame content.
All we have to to is put a link inside of the Turbo Fram, where the requested view *also* includes a Turbo frame with the **same ID**.